packages/client/test/integration/test-node/test/compatibility/compatibility.spec.ts

```diff
@@ -0,0 +1,242 @@
+/**
+ * This file is for Compatibility Matrix tests, which assigns specific
+ * Allure context tags.
+ */
+
+import * as allure from 'allure-vitest'
+import { pipe } from 'fp-ts/function'
+import { describe, expect, TaskContext, test } from 'vitest'
+import { clientFactory, setupPeerTestsLifecycle } from '../util'
+import { datamodel, sugar } from '@iroha2/data-model'
+import { CLIENT_CONFIG } from '@iroha2/test-configuration'
+import { Seq } from 'immutable'
+
+setupPeerTestsLifecycle()
+
+async function annotateAllure(
+    ctx: TaskContext,
+    attrs: { id: number, feature: string, sdkTestId: string, story: string }
+) {
+  await allure.label(ctx, 'allureId', String(attrs.id))
+  await allure.label(ctx, 'feature', attrs.feature)
+  await allure.label(ctx, 'sdk_test_id', attrs.sdkTestId)
+  await allure.label(ctx, 'story', attrs.story)
+  await allure.owner(ctx, 'dulger')
+  await allure.label(ctx, 'permission', 'no_permission_required')
+  await allure.label(ctx, 'sdk', 'Java Script')
+}
+// Read Allure API: https://allurereport.org/docs/vitest-reference/
+describe('Compatibility Matrix tests', () => {
+  // Based on https://github.com/AlexStroke/iroha-java/blob/007a9ac55991cd8a2b519e62a10144156d9f8301/modules/client/src/test/kotlin/jp/co/soramitsu/iroha2/InstructionsTest.kt#L134
+  test('Register domain', async (ctx) => {
+    await annotateAllure(ctx, {
+      id: 4075,
+      feature: 'Domain',
+      story: 'Account registers a domain',
+      sdkTestId: 'register_domain',
+    })
+
+    const DOMAIN_NAME = 'new_domain_name'
+    let result
+    let final
+
+    const { pre, client, getBlocksListener } = clientFactory()
+    const blocks = await getBlocksListener()
+    await allure.step(ctx, "Account registers a new domain with name 'new_domain_name'", async () => {
+      await blocks.wait(async () => {
+        await client.submitExecutable(
+          pre,
+          pipe(sugar.identifiable.newDomain(DOMAIN_NAME), sugar.instruction.register, sugar.executable.instructions),
+        )
+      })
+    })
+    await allure.step(ctx, "Creating 'result' variable wich consist DomainId of created domain", async () => {
```

Comment:
```suggestion
    await allure.step(ctx, "Ensure domain is registered with `FindDomainById` query", async () => {
```

I would merge this step with the next one (with assertions)

Reply:
ready